runtime/type_test.go

```diff
@@ -90,3 +91,92 @@ func TestRuntimeTypeStorage(t *testing.T) {
 
 	assert.Equal(t, `"Int"`, loggedMessage)
 }
+
+func TestBlockTimestamp(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("transaction", func(t *testing.T) {
+
+		t.Parallel()
+
+		runtime := NewInterpreterRuntime()
+
+		script := []byte(`
+			transaction {
+				prepare() {
+					let block = getCurrentBlock()
+					var ts: UFix64 = block.timestamp
+					log(ts.isInstance(Type<UFix64>()))
+
+					var div: UFix64 = 4.0
+
+					// Shouldn't panic
+					var res = ts/div
+				}
+			}
+        `)
+
+		var loggedMessage string
+
+		runtimeInterface := &testRuntimeInterface{
+			getSigningAccounts: func() ([]Address, error) {
+				return nil, nil
+			},
+			log: func(message string) {
+				loggedMessage = message
+			},
+		}
+
+		nextTransactionLocation := newTransactionLocationGenerator()
+		err := runtime.ExecuteTransaction(
+			Script{
+				Source: script,
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  nextTransactionLocation(),
+			},
+		)
+		require.NoError(t, err)
+		assert.Equal(t, "true", loggedMessage, "Block.timestamp is not UFix64")
+
+	})
+
+	t.Run("script", func(t *testing.T) {
+
+		t.Parallel()
+
+		runtime := NewInterpreterRuntime()
+
+		script := []byte(`
+			pub fun main(): [UFix64] {
+				let block = getCurrentBlock()
+				var ts: UFix64 = block.timestamp
+
+				var div: UFix64 = 4.0
+
+				// Shouldn't panic
+				var res = ts/div
+
+				return [ts, res]
+			}
+        `)
+
+		value, err := runtime.ExecuteScript(
+			Script{
+				Source: script,
+			},
+			Context{
+				Interface: &testRuntimeInterface{},
+				Location:  common.ScriptLocation{},
+			},
+		)
+
+		require.NoError(t, err)
+		require.NotNil(t, value)
+		values := value.(cadence.Array).Values
```

Comment:
The type cast can fail, please add type assertion before. (with `require`)

Also, the indexing below can fail, please add an assertion that the array length is 2.